lib/fibonacci.rb

```diff
@@ -1,9 +1,22 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n))
 # Hint, you may want a recursive helper method
 # Hint:  Fib(0) = 0, Fib(1) = 1, work up from there
+
 def fibonacci(n)
```

Comment:
This works, but it's O(n) in space complexity, can you figure out a way to reduce it so that you don't need the entire array?